vms/shift/tests/test_unit.py

```diff
@@ -410,3 +412,166 @@ def test_model_representation(self):
+        edit_request_in_db.start_time = start
+        edit_request_in_db.end_time = end
+        edit_request_in_db.volunteer_shift.volunteer = volunteer_in_db
+        edit_request_in_db.save()
+
+        self.assertEqual(len(EditRequest.objects.all()), 1)
+
+        edit_request_in_db = EditRequest.objects.get(Q(volunteer_shift=vol_shift_in_db))
+
+        # verify correctness
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.start_time), edit_request.volunteer_shift.shift.start_time)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.end_time), edit_request.volunteer_shift.shift.end_time)
+        self.assertEqual(edit_request_in_db.start_time, start)
+        self.assertEqual(edit_request_in_db.end_time, end)
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.first_name, 'Prince')
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.last_name, 'Vegeta')
+
+    def test_model_delete(self):
+        shift = self.create_shift()
+        volunteer = self.create_volunteer()
+
+        edit_request = self.create_edit_request(volunteer, shift)
+
+        # Check db instance creation
+        self.assertEqual(len(EditRequest.objects.all()), 1)
+
+        shift_in_db = Shift.objects.get(Q(job=self.job))
+        volunteer_in_db =  Volunteer.objects.get(Q(first_name='Son'))
+        vol_shift_in_db = VolunteerShift.objects.get(Q(shift=shift_in_db))
+        edit_request_in_db = EditRequest.objects.get(Q(volunteer_shift=vol_shift_in_db))
+
+        # verify correctness
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.first_name, edit_request.volunteer_shift.volunteer.first_name)
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.last_name, edit_request.volunteer_shift.volunteer.last_name)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.start_time), edit_request.volunteer_shift.shift.start_time)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.end_time), edit_request.volunteer_shift.shift.end_time)
+        self.assertEqual(edit_request_in_db.start_time, edit_request.start_time)
+        self.assertEqual(edit_request_in_db.end_time, edit_request.end_time)
+
+        edit_request_in_db.delete()
+        self.assertEqual(len(EditRequest.objects.all()), 0)
+
+    def test_model_representation(self):
+        shift = self.create_shift()
+        volunteer = self.create_volunteer()
+
+        edit_request = self.create_edit_request(volunteer, shift)
+
+        # Check db instance creation
+        self.assertEqual(len(EditRequest.objects.all()), 1)
+
+        shift_in_db = Shift.objects.get(Q(job=self.job))
+        vol_shift_in_db = VolunteerShift.objects.get(Q(shift=shift_in_db))
+        edit_request_in_db = EditRequest.objects.get(Q(volunteer_shift=vol_shift_in_db))
+
+        # Check correctness
+        self.assertEqual(str(edit_request_in_db), 'EditRequest object')
```

Comment:
We should define a `__str__` in model I guess.

Reply:
added